src/zope/app/testing/placelesssetup.py

```diff
@@ -26,13 +25,11 @@
 from zope.app.testing import ztapi
 from zope.container.testing import PlacelessSetup as ContainerPlacelessSetup
 
-class PlacelessSetup(CAPlacelessSetup,
-                     EventPlacelessSetup,
+class PlacelessSetup(EventPlacelessSetup,
                      I18nPlacelessSetup,
                      ContainerPlacelessSetup):
 
     def setUp(self, doctesttest=None):
-        CAPlacelessSetup.setUp(self)
         EventPlacelessSetup.setUp(self)
         ContainerPlacelessSetup.setUp(self)
         I18nPlacelessSetup.setUp(self)
```

Comment:
I wonder if the same problem (`I18nPlacelessSetup.setUp()` now unexpectedly clearing the component registry and everything will affect other packages which use it, such as `zope.formlib` and `Zope`.

```
mg@platonas: ~/src/zopefoundation $ ag zope.i18n.testing
zope.app.testing/src/zope/app/testing/placelesssetup.py
21:from zope.i18n.testing import PlacelessSetup as I18nPlacelessSetup

zope.formlib/src/zope/formlib/tests/functionalsupport.py
15:from zope.i18n.testing import setUp as i18nSetUp

zope.formlib/src/zope/formlib/tests/test_formlib.py
22:from zope.i18n.testing import setUp as i18nSetUp

Zope/src/Testing/ZopeTestCase/placeless.py
19:from zope.i18n.testing import PlacelessSetup as I18nPlacelessSetup
```

Reply:
Looks like zope.formlib is fine (it calls zope.component.testing.setUp just before zope.i18n.testing.setUp).

Reply:
Zope, meanwhile, calls I18nPlacelessSetup.setUp() after CAPlacelessSetup.setUp(self), EventPlacelessSetup.setUp(self), and ContainerPlacelessSetup.setUp(self).

I wouldn't be surprised if it also suffers from the MRO problem, since Testing.ZopeTestCase.placeless.PlacelessSetup looks very much like the one in zope.app.testing.

Reply:
Yeah, apparently it does: https://github.com/zopefoundation/zope.i18n/issues/31

OK, so scratch this PR and I'll make a PR to revert the change in zope.i18n.

Reply:
Thank you!